sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ThroughputProperties.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.Offer;
+import com.azure.cosmos.implementation.OfferAutoscaleSettings;
+
+/**
+ * Represents throughput of the resources in the Azure Cosmos DB service.
+ */
+public class ThroughputProperties {
+    private final Offer offer;
+
+    ThroughputProperties(Offer offer) {
+        this.offer = offer;
+    }
+
+    /**
+     * Create fixed throughput properties.
+     *
+     * @param throughput the throughput
+     * @return the throughput properties
+     */
+    public static ThroughputProperties createFixedThroughput(int throughput) {
+        return new ThroughputProperties(Offer.createFixedOffer(throughput));
+    }
+
+    /**
+     * Create auto scale throughput properties.
+     *
+     * @param maxAutoScaleThroughput the max auto scale throughput
+     * @param autoUpgradethroughputIncrementPercentage the auto upgrade max throughput increment percentage
+     * @return the throughput properties
+     */
+    public static ThroughputProperties createAutoScaledProvisionedThroughput(
```

Comment:
How about excluding Provisioned from the name?
@j82w can you please check for consistency with .NET contracts as well.

Reply:
Its V4 we have a choice to improvise as well.

Reply:
Makes sense. It is createAutoScaledThroughput now.

Reply:
@deborahc do you agree with this?

Reply:
How about: createAutoscaleThroughput for consistency. (the official autoscale name doesn't have the S in scale capitalized).

Reply:
Oh yeah! I should lowercase the S. createAutoscaleThroughput should be final right?

Reply:
changed